File: packages/ios/Sources/Components/Sidebar/Sidebar.figma.swift
import SwiftUI
import Figma

struct KozmosSidebarBasicConnect: FigmaConnect {
    let component = KozmosSidebar<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6807"
    var variant = ["Content": "Basic"]

    var body: some View {
        KozmosSidebar(
            variant: .expanded,
            header: {
                Text("Header")
            },
            navigation: {
                AnyView(Text("Navigation"))
            },
            tools: {
                EmptyView()
            },
            footer: {
                Text("Footer")
            }
        )
    }
}

struct KozmosSidebarSectionsConnect: FigmaConnect {
    let component = KozmosSidebar<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6807"
    var variant = ["Content": "Sections"]

    var body: some View {
        KozmosSidebar(
            variant: .expanded,
            header: {
                Text("Header")
            },
            navigation: {
                AnyView(
                    VStack(alignment: .leading) {
                        Text("Section")
                        Text("Navigation")
                    }
                )
            },
            tools: {
                EmptyView()
            },
            footer: {
                Text("Footer")
            }
        )
    }
}

struct KozmosSidebarToolsConnect: FigmaConnect {
    let component = KozmosSidebar<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6807"
    var variant = ["Content": "Tools"]

    var body: some View {
        KozmosSidebar(
            variant: .expanded,
            header: {
                Text("Header")
            },
            navigation: {
                AnyView(
                    VStack(alignment: .leading) {
                        Text("Section")
                        Text("Navigation")
                    }
                )
            },
            tools: {
                Text("Tools")
            },
            footer: {
                Text("Footer")
            }
        )
    }
}

struct KozmosSidebarRailConnect: FigmaConnect {
    let component = KozmosSidebar<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6807"
    var variant = ["Content": "Rail"]

    var body: some View {
        KozmosSidebar(
            variant: .rail,
            header: {
                Text("Header")
            },
            navigation: {
                AnyView(Text("Navigation"))
            },
            tools: {
                EmptyView()
            },
            footer: {
                Text("Footer")
            }
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/Sidebar/Sidebar.figma.kt
package com.kozmos.components.sidebar

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=752-6807")
class KozmosSidebarConnect {
    @FigmaProperty(FigmaType.Enum, "Content")
    val content: String = Figma.mapping(
        "Basic" to "basic",
        "Sections" to "sections",
        "Tools" to "tools",
        "Rail" to "rail"
    )

    @Composable
    fun ComponentExample() {
        KozmosSidebar(
            variant = if (content == "rail") KozmosSidebarVariant.Rail else KozmosSidebarVariant.Expanded,
            header = {
                Text("Header")
            },
            navigation = {
                if (content == "sections" || content == "tools") {
                    Text("Section")
                }
                Text("Navigation")
            },
            tools = {
                if (content == "tools") {
                    Text("Tools")
                }
            },
            footer = {
                Text("Footer")
            }
        )
    }
}
